### PR TITLE
Implement histogram

### PR DESCRIPTION
**Pull request by unnonouno**
This PR implements histogram method. It only supports int and ndarray as `bins` argument.
I took over #298

### CODE REVIEW

File: cupy/statistics/histogram.py [hunk 1]
```diff
@@ -3,7 +3,70 @@
...
+            max_value = 1.0
+        else:
+            min_value = float(x.min())
+            max_value = float(x.max())
+        if min_value == max_value:
+            min_value -= 0.5
+            max_value += 0.5
+        bins = cupy.linspace(min_value, max_value, bins + 1)
+    elif isinstance(bins, cupy.ndarray):
+        if cupy.any(bins[:-1] > bins[1:]):
```
Comment (okuta) [inline cupy/statistics/histogram.py]: Please use `>=`. See also [here](https://github.com/numpy/numpy/blob/v1.13.0/numpy/lib/function_base.py#L786).

Comment (unnonouno) [reply cupy/statistics/histogram.py]: I referred to this code: https://github.com/numpy/numpy/blob/master/numpy/lib/histograms.py#L315

Comment (okuta) [reply cupy/statistics/histogram.py]: Yes, you are right. But, old NumPy is a different behavior.

Comment (unnonouno) [reply cupy/statistics/histogram.py]: If its behavior changed, we need to skip test with old NumPys.

Comment (okuta) [reply cupy/statistics/histogram.py]: OK, Please fix test?

Comment (unnonouno) [reply cupy/statistics/histogram.py]: I found that this behavior is a bug which occurs with unsigned integer in old numpy and is fixed in numpy 1.13.2.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**